skimage/morphology/tests/test_flood_fill.py

```diff
@@ -0,0 +1,164 @@
+            flood_fill(image, 0, 1)
+
+
+def test_overrange_tolerance_int():
+    image = np.arange(256, dtype=np.uint8).reshape((8, 8, 4))
+    expected = np.zeros_like(image)
+
+    output = flood_fill(image, (7, 7, 3), 0, tolerance=379)
+
+    np.testing.assert_equal(output, expected)
+
+
+def test_overrange_tolerance_float():
+    max_value = np.finfo(np.float32).max
+    min_value = np.finfo(np.float32).min
+
+    image = np.random.uniform(size=(64, 64), low=-1., high=1.).astype(
+        np.float32)
+    image *= max_value
+
+    expected = np.ones_like(image)
+    output = flood_fill(image, (0, 1), 1., tolerance=max_value * 10)
+
+    np.testing.assert_equal(output, expected)
+
+
+def test_inplace_int():
+    image = np.array([[0, 0, 0, 0, 0, 0, 0],
+                      [0, 1, 1, 0, 2, 2, 0],
+                      [0, 1, 1, 0, 2, 2, 0],
+                      [1, 0, 0, 0, 0, 0, 3],
+                      [0, 1, 1, 1, 3, 3, 4]])
+
+    flood_fill(image, (0, 0), 5, inplace=True)
+
+    expected = np.array([[5, 5, 5, 5, 5, 5, 5],
+                         [5, 1, 1, 5, 2, 2, 5],
+                         [5, 1, 1, 5, 2, 2, 5],
+                         [1, 5, 5, 5, 5, 5, 3],
+                         [5, 1, 1, 1, 3, 3, 4]])
+
+    np.testing.assert_array_equal(image, expected)
+
+
+def test_inplace_float():
+    image = np.array([[0, 0, 0, 0, 0, 0, 0],
+                      [0, 1, 1, 0, 2, 2, 0],
+                      [0, 1, 1, 0, 2, 2, 0],
+                      [1, 0, 0, 0, 0, 0, 3],
+                      [0, 1, 1, 1, 3, 3, 4]], dtype=np.float32)
+
+    flood_fill(image, (0, 0), 5, inplace=True)
+
+    expected = np.array([[5., 5., 5., 5., 5., 5., 5.],
+                         [5., 1., 1., 5., 2., 2., 5.],
+                         [5., 1., 1., 5., 2., 2., 5.],
+                         [1., 5., 5., 5., 5., 5., 3.],
+                         [5., 1., 1., 1., 3., 3., 4.]], dtype=np.float32)
+
+    np.testing.assert_allclose(image, expected)
+
+
+def test_inplace_noncontiguous():
+    image = np.array([[0, 0, 0, 0, 0, 0, 0],
+                      [0, 1, 1, 0, 2, 2, 0],
+                      [0, 1, 1, 0, 2, 2, 0],
+                      [1, 0, 0, 0, 0, 0, 3],
+                      [0, 1, 1, 1, 3, 3, 4]])
+
+    # Transpose is noncontiguous
+    image2 = image[::2, ::2]
+
+    flood_fill(image2, (0, 0), 5, inplace=True)
+
+    # The inplace modified result
+    expected2 = np.array([[5, 5, 5, 5],
+                          [5, 1, 2, 5],
+                          [5, 1, 3, 4]])
+
+    np.testing.assert_allclose(image2, expected2)
+
+    # Projected back through the view, `image` also modified
+    expected = np.array([[5, 0, 5, 0, 5, 0, 5],
+                         [0, 1, 1, 0, 2, 2, 0],
+                         [5, 1, 1, 0, 2, 2, 5],
+                         [1, 0, 0, 0, 0, 0, 3],
+                         [5, 1, 1, 1, 3, 3, 4]])
+
+    np.testing.assert_allclose(image, expected)
+
+
+def test_1d():
```

Comment:
Could you add at least one test for the >= 3D case?

Reply:
When there isn't loss of generality I stay 2D, because it's easier for future maintenance (embedding 3D expected results in the test suite is rarely intuitive or easy to parse).

Reply:
I can't follow that line of reasoning yet. In this case the ability to generalize between 2D and ND is an implementation detail of this particular algorithm. I feel like tests shouldn't loose or gain value if the internal implementation changes without a change to the API. If we replaced the algorithm in the future with one were this generalization doesn't hold true we'd have a gap in our test suite were the 3D case is not properly covered.

However I agree that tests for >2D are generally harder to read. Therefore how about a compromise that adds one basic test that can act as a canary for if that generalization is lost?

E.g.:

```python
cube = np.zeros((5, 5, 5))
cube[1:-1, 1:-1, 1:-1] = 1  # sum is 27
filled = flood_fill(cube, (2, 2, 2), 2)
assert filled.sum() == 54
assert_equal(filled[1:-1, 1:-1, 1:-1], np.ones((3, 3, 3)) * 2)
```

I guess I'd prioritize functionality over readability in tests but I recognize that as an opinion which you don't have to share. Please, feel free to mark this as resolved if you don't see the merit in my suggestion. :wink: